### PR TITLE
Add future imports

### PR DESCRIPTION
**Pull request by whit537**
This adds `__future__` imports to `c7n`, `tests`, and `tools/c7n_{mailer,sentry}`. Step towards #705. Tested mailer and sentry manually per their READMEs.

### CODE REVIEW

**Comment by kapilt** [comment]
this seems to be blind application of future imports at random, are all of them needed?

**Comment by whit537** [comment]
Blorg ... [failure](https://travis-ci.org/capitalone/cloud-custodian/builds/235194129#L682-L683) is from flake8 re: lambda assignment.

> this seems to be blind application of future imports at enmasse, are all of them needed?

Yeah, that's a pretty standard way to start a Python 3 port, I think, though [per Instagram](https://www.youtube.com/watch?v=66XoCk79kjM#t=29m57s) the conservative option is to only apply `absolute_import` and `print_function` due to subtle behavior differences in Python 2 with `division` and `unicode_literals`. Would you prefer the conservative option? I can also try to limit to what's needed per file, but not sure what's driving that as it means more work and the only downside I'm seeing to blanket application is mild boilerplate profusion.

File: c7n/__init__.py [hunk 1]
```diff
@@ -12,3 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from __future__ import absolute_import, division, print_function, unicode_literals
```
Comment (kapilt) [inline c7n/__init__.py]: empty files even?

File: c7n/version.py [hunk 2]
```diff
@@ -11,5 +11,6 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
```
Comment (kapilt) [inline c7n/version.py]: really shouldn't be here either, this file is a constant variable def, we may gen in the future, we can make it explicit on u"" but i'd rather not carry imports here.

Comment (whit537) [inline c7n/version.py]: Backed out in 09f6211958693d1a3296a03f651aea9efa1a70a4. Will look at empties as well ...

File: c7n/__init__.py (hunk 1 above)
Comment (whit537) [inline c7n/__init__.py]: Backed out in 1b076ac6d3b8bbb76bad3e9ce221fab1690ef475.

**Comment by whit537** [comment]
Rebased to resolve conflicts and squashed into b3fb08677b2d80fa35e94982af28bc53da672607. Was 1b076ac6d3b8bbb76bad3e9ce221fab1690ef475.

**Comment by whit537** [comment]
Topped up, ready for rereview @kapilt.

**Comment by whit537** [comment]
Rebased, was b3fb08677b2d80fa35e94982af28bc53da672607.

**Comment by whit537** [comment]
Rebased, was a8324add59b79d3e33d9b36e6da950cd3f489a0b.

**Comment by whit537** [comment]
Rebased, was a8324add59b79d3e33d9b36e6da950cd3f489a0b.

**Comment by whit537** [comment]
Alright, Drone is green.

**Comment by kapilt** [comment]
still showing some conflicts but that looks like from early am today around the same time.

**Comment by kapilt** [comment]
resolved through web ui.

**Comment by kapilt** [comment]
albeit failing now on drone ci

**Comment by whit537** [comment]
I can debug tomorrow AM if helpful.

**Comment by whit537** [comment]
Rebased, was d7bdcbf03faa56b2899afa456125c71a99f2f853. What say you, Drone?

**Comment by whit537** [comment]
Green! Not sure where Travis is, though ...

**Comment by whit537** [comment]
@kapilt Any idea why Travis isn't building this PR? I can debug if you don't have a quick answer ...

**Comment by kapilt** [comment]
its unclear why travis isn't running on this branch, i don't see it on https://travis-ci.org/capitalone/cloud-custodian/pull_requests

possibly an issue with the .travis.yml file in this branch?  it seems to work with some of the followup branches though, its a bit unclear.

**Comment by whit537** [comment]
Okay. There are no `.travis.yml` changes here. Hunch so far is it's related to me running Travis against this branch in my fork before creating a PR here? Looking into it ...

**Comment by whit537** [comment]
Rebased, was 105a919623a66a71eda07b2fa9f6fe0e9197a793.

**Comment by whit537** [comment]
Yeah, I think that was it. Travis building and reporting to Coveralls now, will watch in future. However, Drone is now reporting failure:

``
$ git fetch --no-tags origin +refs/pull/1214/merge:
error: RPC failed; result=18, HTTP code = 200
fatal: The remote end hung up unexpectedly
```

Restarted a few times. Network failure from Drone box? Command works locally:

```
$  git fetch --no-tags upstream +refs/pull/1214/merge
remote: Counting objects: 1, done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (1/1), done.
From https://github.com/capitalone/cloud-custodian
 * branch            refs/pull/1214/merge -> FETCH_HEAD
$
```

**Comment by whit537** [comment]
Travis and Coveralls working. Drone is not, but it's failing so early I don't see a way to debug via `.drone.yml` changes, not sure what else I have access to. Furthermore Drone is passing on https://github.com/capitalone/cloud-custodian/pull/1234 which includes this so I suggest we focus over there.

**Comment by whit537** [comment]
Drone still failing afaict.

**Comment by whit537** [comment]
Per call @kapilt restarted Drone and of course it passsed. 😜

**Pull request merged**